### PR TITLE
DAOS-5868 mgmt: pool destroy corpcs to inclusive rank list

### PR DESCRIPTION
**Pull request by kccain**
With this change, pool destroy corpcs to destroy the pool service,
and to destroy all pool targets, are sent to an explicit list
of ranks. The list of ranks for pool service is already available
in the svc_ranks argument to ds_mgmt_destroy_pool(). A separate CaRT
RPC POOL_RANKS_GET is added in this change, issued by the mgmt
ioserver and handled by the PS leader ioserver. The returned rank
list is then used as the destination for the MGMT_TGT_DESTROY corpc.

Before this change, an excluded list of pool ranks was produced,
based on status MAP_RANKS_DOWN, and a corpc was to all cart primary
group ranks except those in the excluded list. This caused the
corpcs to be directed to system nodes not involved with the pool,
and that may have been independently stopped - in which case
pool destroy corpcs would experience errors such as DER_UNREACH.

Enhance TestPool.start_rebuild() with a boolean argument "force" to
specify whether the --force= argument is to be used with dmg system stop.
And update rebuild/pool_destroy_rebuild.py test to specify force=True when
stopping the rank to trigger rebuild.
    
Enhance TestPool.wait_for_rebuild() with a "smart sleep" to query the pool
periodically until a successful reply, before then monitoring for the
rebuild status to know that rebuild has started or completed.

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

**Review by liw — APPROVED** [review]
Looks good to me. (See also my comment today on DAOS-5868.)

File: src/pool/srv_pool.c
Comment (liw) [inline src/pool/srv_pool.c:5074]: Agree with your concern (on Skype) that this is potentially racy (due to trying to hold pool->sp_lock). Even if we were to fix that by reading from the DB, by the time the reply reaches the MS rank, the pool map may still have changed. So this is an inherent disadvantage of our approach here.

File: src/include/daos_srv/iv.h
Comment (kccain) [inline src/include/daos_srv/iv.h:34]: Just to be clear, this change I proactively merged from  https://github.com/daos-stack/daos/pull/4564/  before that PR was landed to master (it has since landed but it is not worth merging master and running through CI testing in this patch).

File: src/rebuild/srv.c
Comment (kccain) [inline src/rebuild/srv.c:1311]: Just to be clear, this change I proactively merged from  https://github.com/daos-stack/daos/pull/4564/  before that PR was landed to master (it has since landed but it is not worth merging master and running through CI testing in this patch).

**Comment by kprantis** [comment]
This is one of our few remaining 1.1.3 tickets, if reviewers/gatekeepers could please prioritize accordingly, it would be much appreciated.

**Review by saurabhtandan — APPROVED** [review]
Test side changes LGTM.

**Review by wangdi1 — APPROVED** [review]

File: src/pool/srv_pool.c
Comment (kccain) [reply src/pool/srv_pool.c:5082]: Yes I see. This came up in another change that I made a while ago (list containers). I think here, and then, I just used the existing examples in this file as a model (e.g., ds_pool_svc_query). For this PR we should leave it consistent. We can experiment subsequently with removing the sleeps for the rc == -DER_TRUNC scenarios.

For the other cases like rc == RSVC_CLIENT_RECHOOSE it may be helpful because there could be a leader transition in progress and it may be helpful to hold off while a raft election is happening in the pool service.

File: src/mgmt/srv_pool.c
Comment (kccain) [reply src/mgmt/srv_pool.c:310]: I understand this has come up as an idea in the past. I have not spent a lot of time thinking about how it would  work and what other challenges it would add. I'm sorry at the moment I can't remember some of the detailed points. Certainly not feasible for me to implement it in this patch.

**Review by liw — APPROVED** [review]

**Pull request merged**
